packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -143,6 +157,23 @@ class Dismissible extends StatefulWidget {
   /// and [onDismissed] callbacks will not run.
   final ConfirmDismissCallback? confirmDismiss;
 
+  /// Gives the app the ability to accept or reject a dismiss gesture, overriding
+  /// the default behavior.
+  ///
+  /// Returning `null` will use de default behavior.
+  ///
+  /// {@tool snippet}
+  /// The default flinging effect can be disabled by using:
+  /// ```dart
+  /// Dismissible(
+  ///   key: const ValueKey<String>('dismissible-key'),
+  ///   shouldDismiss: (AcceptDismissDetails details) => details.reached ? null : false,
+  ///   child: const ListTile(title: Text('Slide me to dismiss')),
+  /// )
+  /// ```
+  /// {@end-tool}
+  final AcceptDismissCallback? shouldDismiss;
```

Comment:
I think the concerns you mentioned earlier about confusion between `shouldDismiss` and `confirmDismiss` are valid.

What do you think about the following name change?

```dart
// before
final AcceptDismissCallback? shouldDismiss;
class AcceptDismissDetails {}


// after
final TriggerDismissCallback? shouldTriggerDismiss;
class TriggerDismissDetails {}
```